pkg/functions/handlers.go

```diff
@@ -59,23 +63,76 @@ func ConfigureHandlers(api middleware.RoutableAPI, h Handlers) {
 }
 
 type defaultHandlers struct {
-	backend        backend.Backend
-	httpClient     *http.Client
-	namespace      string
-	imageRegistry  string
-	sourceRootPath string
-	imagesClient   client.ImagesClient
+	backend       backend.Backend
+	httpClient    *http.Client
+	namespace     string
+	imageRegistry string
+	storageConfig *config.StorageConfig
+	imagesClient  client.ImagesClient
 }
 
 // NewHandlers is the constructor for the function manager API knHandlers
-func NewHandlers(kubeconfPath, namespace, imageRegistry, sourceRootPath string, imagesClient client.ImagesClient) Handlers {
+func NewHandlers(kubeconfPath, namespace, imageRegistry, ingressGateway, buildImage string, storageConfig *config.StorageConfig, imagesClient client.ImagesClient) Handlers {
```

Comment:
This is getting a little long in the number of parameters. Would it make sense to combine some of these into an object that provides operations on related parameters? From here it looks like this would require significant changes to the defaultHandlers struct, so it may not be feasible.